python/samples/concepts/memory/azure_ai_search_as_plugin.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+
+import asyncio
+from collections.abc import Coroutine
+from typing import Annotated, Any
+
+from pydantic import BaseModel
+
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.function_choice_behavior import FunctionChoiceBehavior
+from semantic_kernel.connectors.ai.open_ai import (
+    OpenAIChatCompletion,
+    OpenAIChatPromptExecutionSettings,
+    OpenAIEmbeddingPromptExecutionSettings,
+    OpenAITextEmbedding,
+)
+from semantic_kernel.connectors.memory.azure_ai_search import AzureAISearchCollection
+from semantic_kernel.contents import ChatHistory
+from semantic_kernel.data import (
+    VectorSearchFilter,
+    VectorSearchOptions,
+    VectorStoreRecordDataField,
+    VectorStoreRecordKeyField,
+    VectorStoreRecordUtils,
+    VectorStoreRecordVectorField,
+    vectorstoremodel,
+)
+from semantic_kernel.data.search_options import SearchOptions
+from semantic_kernel.data.text_search.vector_store_text_search import VectorStoreTextSearch
+from semantic_kernel.filters.filter_types import FilterTypes
+from semantic_kernel.filters.functions.function_invocation_context import FunctionInvocationContext
+from semantic_kernel.functions import (
+    KernelArguments,
+    KernelParameterMetadata,
+)
+
+
+@vectorstoremodel
```

Comment:
This class and any other concept or sample would benefit from a lot more comments describing what is happening and why, and also helping the reader understand which parts are important to learn the new concepts and which parts are just noise.

Reply:
I second this. I almost feel like a "getting started with memory" folder that logically walks users/devs through the new abstractions from easy to more advanced could make sense here.

Reply:
restructured this one, I think a full getting started folder is not needed, but made sure it is easier to follow now.